nixos/modules/services/security/openbao.nix

```diff
@@ -0,0 +1,140 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.openbao;
+
+  settingsFormat = pkgs.formats.json { };
+in
+{
+  options = {
+    services.openbao = {
+      enable = lib.mkEnableOption "OpenBao daemon";
+
+      package = lib.mkPackageOption pkgs "openbao" { };
+
+      settings = lib.mkOption {
+        description = ''
+          Settings of OpenBao.
+
+          See [documentation](https://openbao.org/docs/configuration) for more details.
+        '';
+        example = lib.literalExpression ''
+          {
+            listener.tcp = {
+              tls_acme_email = config.security.acme.defaults.email;
+              tls_acme_domains = [ "example.com" ];
+              tls_acme_disable_http_challenge = true;
+            };
+
+            cluster_addr = "http://127.0.0.1:8201";
+            api_addr = "https://example.com";
+
+            storage.raft = {
+              path = "/var/lib/openbao";
+              node_id = config.networking.hostName;
+            };
+          }
+        '';
+        type = lib.types.submodule {
+          freeformType = settingsFormat.type;
+          options = {
+            listener.tcp = {
```

Comment:
this should probably be an array - it's possible to declare multiple listeners in Bao: https://openbao.org/docs/configuration/listener/tcp/#listening-on-multiple-interfaces

Reply:
Yes, I realized, I'm thinking of how to do it properly.

Reply:
Actually, it uses an attrset with the "type" tag field, see my updated implementation.